autokeras/hypermodel/processor.py

```diff
@@ -111,3 +111,164 @@ def transform(self, hp, data):
+        self.batch_num = 0
+        self.target_height = 0
+        self.target_width = 0
+        self.channels = 0
+
+    def fit(self, hp, inputs):
+        if self.first:
+            self.first = False
+            inputs = tf.convert_to_tensor(inputs)
+            self.length_dim = len(inputs.shape)
+            if self.length_dim != 4:
+                raise ValueError(
+                    'The input of x_train should be a [batch_size, height, '
+                    'width, channel] '
+                    'shape tensor or list, but we get %s' % inputs.shape)
+            (self.batch_num, self.target_height,
+             self.target_width, self.channels) = inputs.shape
+            # TODO: Set the arguments if user didn't set
+            return inputs
+        else:
+            pass
+
+    @staticmethod
+    def __get_min_and_max(value, name):
+        if isinstance(value, (tuple, list)):
+            if len(value) != 2:
+                raise ValueError(
+                    'Argument %s expected either a float between 0 and 1, '
+                    'or a tuple of 2 floats between 0 and 1, '
+                    'but got: %s' % (value, name))
+            min_value, max_value = value
+        else:
+            min_value = 1. - value
+            max_value = 1. + value
+        return min_value, max_value
+
+    @staticmethod
+    def guassian_noise(image):
+        noise = tf.random.normal(shape=tf.shape(image),
+                                 mean=0.0, stddev=1.0, dtype=tf.float32)
+        image = tf.add(image, noise)
+        return image
+
+    @staticmethod
+    def translation(image, hp):
+        x = tf.image.pad_to_bounding_box(image, hp.Choice('translation_top'),
+                                         hp.Choice('translation_left'),
+                                         hp.Choice('target_height') +
+                                         hp.Choice('translation_bottom') +
+                                         hp.Choice('translation_top'),
+                                         hp.Choice('target_width') +
+                                         hp.Choice('translation_right') +
+                                         hp.Choice('translation_left'))
+        image = tf.image.crop_to_bounding_box(x, hp.Choice('translation_bottom'),
+                                              hp.Choice('translation_right'),
+                                              hp.Choice('target_height'),
+                                              hp.Choice('target_width'))
+        return image
+
+    @staticmethod
+    def rotation(image, hp):
+        if hp.Choice('rotation_range') == 90:
+            image = tf.image.rot90(image, k=1)
+        elif hp.Choice('rotation_range') == 180:
+            image = tf.image.rot90(image, k=2)
+        elif hp.Choice('rotation_range') == 270:
+            image = tf.image.rot90(image, k=3)
+        else:
+            image = tf.image.rot90(image, k=4)
+        return image
```

Comment:
you may want something like `image = tf.image.rot90(image, k=hp.Choice('rotation_range') / 90)`